src/charm.py

```diff
@@ -283,9 +283,15 @@ def _configure(self):  # noqa: C901
             return
 
         if restart:
-            self._container.add_layer(self._name, new_layer, combine=True)
-            self._container.restart(self._name)
-            logger.info("Loki (re)started")
+            try:
+                self._container.add_layer(self._name, new_layer, combine=True)
+                self._container.restart(self._name)
+                logger.info("Loki (re)started")
+            except ChangeError as e:
+                msg = str(e)
```

Comment:
ChangeError has a nice [`__str__`](https://github.com/canonical/operator/blob/f683a0bcc2601daad394e652740061559fa88e13/ops/pebble.py#L351) and we can use it directly:
```suggestion
                msg = f"Failed to restart loki: {e}"  # or e.err?
```